### PR TITLE
ci: apply WSL hanging workaround to MSI builder

### PR DESCRIPTION
**Pull request by austinvazquez**
Issue #, if available:
The MSI builder workflow is stuck due to WSL hanging issue.

*Description of changes:*
This change applies the workaround for WSL hanging to the MSI builder workflow.

*Testing done:*
CI is successful

- [x] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/actions/cleanup-windows-env/action.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    - name: Clean up Finch configuration and artifacts
+      if: inputs.deep-clean
+      shell: pwsh
+      run: |
+        Remove-Item C:\Users\Administrator\.finch -Recurse -ErrorAction Ignore
+        make clean
+        exit 0 # Cleanup may set the exit code e.g. if a file doesn't exist; just ignore
+
+branding:
```
Comment (pendo324) [inline .github/actions/cleanup-windows-env/action.yml:49]: This is cool, didn't know we could customize branding!

File: .github/actions/cleanup-windows-env/action.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        wsl --list --verbose
+
+    - name: Clean up Finch application data
+      shell: pwsh
+      run: |
+        Remove-Item C:\Users\Administrator\AppData\Local\.finch -Recurse -ErrorAction Ignore
+        exit 0 # Cleanup may set the exit code e.g. if a file doesn't exist; just ignore
+
+    - name: Clean up Finch configuration and artifacts
+      if: inputs.deep-clean
```
Comment (pendo324) [inline .github/actions/cleanup-windows-env/action.yml:42]: Seems like we're not using this right now, what's the intention?

**Review by pendo324 — APPROVED** [review]
LGTM. A nice centralization of the Windows cleanup logic. Hopefully your improvements make it a little more successful in cleaning up too

**Comment by austinvazquez** [comment]
Synced with @pendo324 offline but for transparency. GitHub composite actions do not offer full step support when compared to GitHub Actions Job steps. For example, `steps[*].timeout-minutes` and `steps[*].needs` is not supported yet.

Because of these gaps, the composite action would not support what we need quickly and intuitively, so centralization of the WSL workaround would need more bake time. In the short term, let's apply the workaround across the board.

**Review by pendo324 — APPROVED** [review]

**Pull request merged**